accounts/pkg/storage/cs3.go

```diff
@@ -0,0 +1,293 @@
+	if err != nil {
+		return err
+	}
+
+	resp, err := r.dataProvider.get(r.accountURL(id), t)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode == http.StatusNotFound {
+		return &notFoundErr{"account", id}
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if err := json.Unmarshal(b, &a); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteAccount deletes an account via cs3 by id
+func (r CS3Repo) DeleteAccount(ctx context.Context, id string) (err error) {
+	t, err := r.authenticate(ctx)
+	if err != nil {
+		return err
+	}
+
+	ctx = metadata.AppendToOutgoingContext(ctx, token.TokenHeader, t)
+
+	resp, err := r.storageProvider.Delete(ctx, &provider.DeleteRequest{
+		Ref: &provider.Reference{
+			Spec: &provider.Reference_Path{Path: fmt.Sprintf("/meta/%s/%s", accountsFolder, id)},
+		},
+	})
+
+	if err != nil {
+		return err
+	}
+
+	// TODO Handle other error codes?
+	if resp.Status.Code == v1beta11.Code_CODE_NOT_FOUND {
+		return &notFoundErr{"account", id}
+	}
+
+	return nil
+}
+
+// WriteGroup writes a group via cs3 and modifies the provided group (e.g. with a generated id).
+func (r CS3Repo) WriteGroup(ctx context.Context, g *proto.Group) (err error) {
+	t, err := r.authenticate(ctx)
+	if err != nil {
+		return err
+	}
+
+	ctx = metadata.AppendToOutgoingContext(ctx, token.TokenHeader, t)
+	if err := r.makeRootDirIfNotExist(ctx, groupsFolder); err != nil {
+		return err
+	}
+
+	var by []byte
+	if by, err = json.Marshal(g); err != nil {
+		return err
+	}
+
+	_, err = r.dataProvider.put(r.groupURL(g.Id), bytes.NewReader(by), t)
+	return err
+}
+
+// LoadGroup loads a group via cs3 by id and writes it to the provided group
+func (r CS3Repo) LoadGroup(ctx context.Context, id string, g *proto.Group) (err error) {
+	t, err := r.authenticate(ctx)
+	if err != nil {
+		return err
+	}
+
+	resp, err := r.dataProvider.get(r.groupURL(id), t)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode == http.StatusNotFound {
+		return &notFoundErr{"group", id}
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	return json.Unmarshal(b, &g)
+}
+
+// DeleteGroup deletes a group via cs3 by id
+func (r CS3Repo) DeleteGroup(ctx context.Context, id string) (err error) {
+	t, err := r.authenticate(ctx)
+	if err != nil {
+		return err
+	}
+
+	ctx = metadata.AppendToOutgoingContext(ctx, token.TokenHeader, t)
+
+	resp, err := r.storageProvider.Delete(ctx, &provider.DeleteRequest{
+		Ref: &provider.Reference{
+			Spec: &provider.Reference_Path{Path: fmt.Sprintf("/meta/%s/%s", groupsFolder, id)},
+		},
+	})
+
+	if err != nil {
+		return err
+	}
+
+	// TODO Handle other error codes?
+	if resp.Status.Code == v1beta11.Code_CODE_NOT_FOUND {
+		return &notFoundErr{"group", id}
+	}
+
+	return err
+}
+
+func (r CS3Repo) authenticate(ctx context.Context) (token string, err error) {
+	u := &user.User{
+		Id:     &user.UserId{},
```

Comment:
we have to make sure / monitor / write tests for this use case. Reva might change and minting an empty user could break easily in the future.

Reply:
Oh, I'm seeing this already assigned in line 219. Nevermind the previous comment :)